x-pack/plugin/kql/src/test/java/org/elasticsearch/xpack/kql/parser/KqlParserFieldQueryTests.java

```diff
@@ -0,0 +1,311 @@
+            assertMatchQueryBuilder(parseKqlQuery(kqlFieldQuery(fieldName, "OR AND")), fieldName, "OR AND");
+            assertMatchQueryBuilder(parseKqlQuery(kqlFieldQuery(fieldName, "OR OR")), fieldName, "OR OR");
+            assertMatchQueryBuilder(parseKqlQuery(kqlFieldQuery(fieldName, "OR NOT")), fieldName, "OR NOT");
+            assertMatchQueryBuilder(parseKqlQuery(kqlFieldQuery(fieldName, "NOT")), fieldName, "NOT");
+            assertMatchQueryBuilder(parseKqlQuery(kqlFieldQuery(fieldName, "NOT AND")), fieldName, "NOT AND");
+            assertMatchQueryBuilder(parseKqlQuery(kqlFieldQuery(fieldName, "NOT OR")), fieldName, "NOT OR");
+
+            // Check we can use quoted field name as well
+            assertThat(
+                parseKqlQuery(kqlFieldQuery(fieldName, "foo")),
+                equalTo(parseKqlQuery(kqlFieldQuery(quoteString(fieldName), "foo")))
+            );
+        }
+    }
+
+    public void testParseQuotedStringKeywordFieldQuery() {
+        // Single word
+        assertTermQueryBuilder(parseKqlQuery(kqlFieldQuery(KEYWORD_FIELD_NAME, quoteString("foo"))), KEYWORD_FIELD_NAME, "foo");
+
+        // Multiple words
+        assertTermQueryBuilder(parseKqlQuery(kqlFieldQuery(KEYWORD_FIELD_NAME, quoteString("foo bar"))), KEYWORD_FIELD_NAME, "foo bar");
+
+        // Containing unescaped KQL reserved keyword
+        assertTermQueryBuilder(
+            parseKqlQuery(kqlFieldQuery(KEYWORD_FIELD_NAME, quoteString("not foo and bar or baz"))),
+            KEYWORD_FIELD_NAME,
+            "not foo and bar or baz"
+        );
+
+        // Containing unescaped KQL reserved characters
+        assertTermQueryBuilder(
+            parseKqlQuery(kqlFieldQuery(KEYWORD_FIELD_NAME, quoteString("foo*: {(<bar>})"))),
+            KEYWORD_FIELD_NAME,
+            "foo*: {(<bar>})"
+        );
+    }
+
+    public void testParseQuotedStringMatchFieldsQuery() {
+        for (String fieldName : List.of(TEXT_FIELD_NAME, INT_FIELD_NAME, DOUBLE_FIELD_NAME)) {
+
+            // Single word
+            assertMatchPhraseBuilder(parseKqlQuery(kqlFieldQuery(fieldName, quoteString("foo"))), fieldName, "foo");
+
+            // Multiple words
+            assertMatchPhraseBuilder(parseKqlQuery(kqlFieldQuery(fieldName, quoteString("foo bar"))), fieldName, "foo bar");
+
+            // Containing unescaped KQL reserved keyword
+            assertMatchPhraseBuilder(
+                parseKqlQuery(kqlFieldQuery(fieldName, quoteString("not foo and bar or baz"))),
+                fieldName,
+                "not foo and bar or baz"
+            );
+
+            // Containing unescaped KQL reserved characters
+            assertMatchPhraseBuilder(parseKqlQuery(kqlFieldQuery(fieldName, quoteString("foo*: {(<bar>})"))), fieldName, "foo*: {(<bar>})");
+        }
+    }
+
+    public void testParseWildcardMatchFieldQuery() {
+        for (String fieldName : List.of(TEXT_FIELD_NAME, INT_FIELD_NAME, DOUBLE_FIELD_NAME)) {
```

Comment:
just to clarify - do we have the functionality yet to query multiple fields e.g. `datastream.*: logs` - I haven't seen in the tests or on the next TODO list

Reply:
This is actually supported and the tested using:

```java
public void testFieldWildcardFieldQueries() {
    List<String> queries = List.of("foo", "foo bar", quoteString("foo"), "foo*");
    for (String query : queries) {
        for (String fieldNamePattern : List.of("mapped_*", "*")) {
            List<String> searchableFields = searchableFields(fieldNamePattern);
            String kqlQuery = kqlFieldQuery(fieldNamePattern, query);
            BoolQueryBuilder parsedQuery = asInstanceOf(BoolQueryBuilder.class, parseKqlQuery(kqlQuery));
            assertThat(parsedQuery.mustNot(), empty());
            assertThat(parsedQuery.must(), empty());
            assertThat(parsedQuery.filter(), empty());
            assertThat(parsedQuery.minimumShouldMatch(), equalTo("1"));
            assertThat(parsedQuery.should(), hasSize(searchableFields.size()));

            assertThat(
                parsedQuery.should(),
                containsInAnyOrder(searchableFields.stream().map(fieldName -> parseKqlQuery(kqlFieldQuery(fieldName, query))).toArray())
            );
        }
    }
}
```